### DISCUSSION TITLE
分享我的配置：4k屏玩720p的galgame

### DISCUSSION THREAD

**Discussion by XiaofengdiZhu** [Blinue/Magpie#250] [Show and tell]
首先进入右键游戏的exe，打开属性中的兼容性设置，将“高 DPI 缩放替代”设置为“应用程序”
然后会编辑配置文件的应该懂怎么用：
```json
"effects": [
      //先放大四倍
      {
        "effect": "Anime4K_Upscale_Denoise_L"
      },
      {
        "effect": "Anime4K_Upscale_L"
      },
      //再缩放到屏幕分辨率
      {
        "effect": "SSimDownscaler",
        "scale": [ -1, -1 ]
      },
      //最后还原一下动漫画面线条
      {
        "effect": "Anime4K_Restore_VL"
      }
    ]
```
2023-3-27更新4k 144hz玩720p的galgame的配置：
Anime4K_Upscale_VL > Lanczos > Anime4K_Restore_L
RTX3080每帧需要5ms

**Comment by Blinue** [comment]
这配置很 mpv。dev分支已经移植了 Anime4K_GAN，相比 Upscale 细节更多，但有些锐化过度。

如果使用 gdi 和 dwmsharedsurface 捕获模式那么修改“高 DPI 缩放替代”是可选项，对于 VN 我推荐使用 GDI （或 v0.8 即将推出的 Desktop Duplication）

**Reply by XiaofengdiZhu** [reply]
DwmSharedSurface这个捕获方式相比较前两者还有什么优势吗？首页readme几乎没对它介绍

**Reply by Blinue** [reply]
行为基本和 gdi 一致，但显存占用较少。因为这是一个不受微软支持的API，所以有失败的可能

**Reply by XiaofengdiZhu** [reply]
简单试了一下gdi和DwmSharedSurface，不限制帧数直接飙到400帧以上……同时确实用这两个就不用调高dpi缩放替代了。那为什么最推荐Graphics Capture呢？

**Reply by Blinue** [reply]
Graphics Capture 是 3D 游戏的首选捕获方式，它最快，最稳定，兼容性最好。使用 Graphics Capture 时帧率不会高于垂直同步

**Comment by frdoma** [comment]
2次Anime4K+Restore_VL很吃显卡吧，估计得3060以上显卡才能跑得动。
Restore算法感觉主要是用来提升清晰度的，但也会造成和原图的细节进一步的差异化，有时候看起来很惊艳有时候看起来又显得很不自然。不同的游戏可能用不用的配置更好。

**Reply by XiaofengdiZhu** [reply]
3080只有80帧😂

**Comment by Blinue** [comment]
如果要放大4倍的话这里有一个相对更快的配置 https://github.com/Blinue/Magpie/wiki/%E8%87%AA%E5%AE%9A%E4%B9%89%E7%BC%A9%E6%94%BE%E9%85%8D%E7%BD%AE#%E7%A4%BA%E4%BE%8B

**Reply by XiaofengdiZhu** [reply]
单纯为快确实~期待新版用您提到的新Anime4K_GAN算法（懒得装C++环境）

**Comment by rxia24** [comment]
我个人是Anime4K_Upscale_Denoise_UL -> Bilinear 用在480P上，效果可以。

**Reply by XiaofengdiZhu** [reply]
restore效果太好了，不加这个我已经无法接受了

**Comment by lisansas** [comment]
你还别说，效果确实好。

**Comment by AyaRu97** [comment]
更新一下現在用的配置
960p以下的galgame : (感謝chlbc的Bicubic配置 ! )
CuNNy-4x24 NVL > Bicubic (適應0.5倍,B=0,C=0.75) > AdaptiveSharpen(0.05) > CuNNy-4x24 NVL
因為換了120hz螢幕，所以gpu1功耗會比較高，鎖60幀的話大約是90w
<img width="3840" height="2160" alt="bandicam 2025-08-25 05-28-48-526" src="https://github.com/user-attachments/assets/88c81662-cd4d-42d4-b7ce-821857bf1fc7" />
1080p galgame : CuNNy-4x24 NVL > AdaptiveSharpen(0.05)
------------------------------分隔線------------------------------
這樣的配置應該會更好一些，以樓主的顯卡來說應該是跑得動的。
在兩次Upscale之間加一個Bicubic可以減低無謂的性能付出
P.S.兩次VL跑不動的話就先VL再L，再不行就兩次L...如此類推
```
//4k60hz螢幕
"effects": [
//先放大兩倍(1280x720 > 2560x1440)
//原生800x600(800x600 > 1600x1200)
                {
                    "name": "Anime4K\\Anime4K_Upscale_VL"
                },
//選倍數，縮小25%(x,y:0.75)(2560x1440 > 1920x1080)
//原生800x600則縮小10%(x,y:0.9)(1600x1200 > 1440x1080)
                {
                    "name": "Bicubic",
                    "scalingType": 0,
                    "scale": {
                        "x": 0.75,
                        "y": 0.75
                    },
                    "parameters": {
                        "paramB": 0.0,
                        "paramC": 0.75
                    }
                },
//再放大兩倍(1920x1080 > 3840x2160)
//原生800x600(1440x1080 > 2880x2160)
                {
                    "name": "Anime4K\\Anime4K_Upscale_VL"
                },
//還原線條
                {
                    "name": "Anime4K\\Anime4K_Restore_L"
                }
            ]
```
留意功耗 (GPU1 PWR)
Upscale_VL > Bicubic > Upscale_VL > Restore_L (垂直同步+允許延遲)
     1440p    >  1080p  >    2160P
![bandicam 2023-09-15 19-38-19-101](https://github.com/Blinue/Magpie/assets/115739200/55e94ce3-e56e-440c-bee5-232ebd7ae96a)

Upscale_VL > Upscale_VL > Bicubic > Restore_L (垂直同步+允許延遲)
     1440p    >     2880p     >  2160p
![bandicam 2023-09-15 19-38-59-867](https://github.com/Blinue/Magpie/assets/115739200/2c0378d4-becd-4f05-bd85-20342ebef522)

另外，因個人不太喜歡Restore效果..所以我的配置如下(720p>4k)(垂直同步)
```
           "effects": [
//先放大兩倍(720p>1440p)
                {
                    "name": "Anime4K\\Anime4K_Upscale_VL"
                },
//縮小至1080p
                {
                    "name": "Bicubic",
                    "scalingType": 0,
                    "scale": {
                        "x": 0.75,
                        "y": 0.75
                    },
                    "parameters": {
                        "paramB": 0.0,
                        "paramC": 0.75
                    }
                },
//再放大兩倍(1080p>4k)
                {
                    "name": "Anime4K\\Anime4K_Upscale_VL"
                },
//再稍微銳化一下
                {
                    "name": "FSR\\FSR_RCAS",
                    "parameters": {
                        "sharpness": 0.5
                    }
                }
            ]
```
![bandicam 2023-09-15 18-45-17-018](https://github.com/Blinue/Magpie/assets/115739200/aaa8bc91-5b24-4008-ab33-0bb6f8908128)

**Reply by XiaofengdiZhu** [reply]
我一直以为两次放大之间加缩小会降低画质，我拿图片对比软件仔细对比了下……中间先缩小一下反而线条更好

**Reply by AyaRu97** [reply]
對啊，但這配置縮小算法就變得有點重要。
以我的配置來說，用SSIM Oversharp1是災難級的...(SSIM帶有銳化效果..所以刪了FSR_RCAS)
試了不同的縮小算法..個人是比較喜歡 bicubic Photoshop(B=0 C=0.75)(上一個留言最後的圖就是用這的)
↓↓↓↓SSIM Oversharp1↓↓↓↓
![bandicam 2023-09-15 21-27-02-918](https://github.com/Blinue/Magpie/assets/115739200/add90656-fa19-4d25-a7bc-48b3f64d66e3)

**Reply by suzushiikazema** [reply]
非常感谢。。这套算法真的厉害了
之前玩720p一直就是拉2层anime4k_upscale+anime4k_restore，这套画质好但是总感觉色彩很不自然，其他设置效果又很差。没想到中间插一个缩放后效果居然能这么好，显卡也不吵了（之前那么有开300多瓦功耗风扇飞起）。

**Reply by luojunyuan** [reply]
这套配置在我的m3pro上执行有30帧

**Reply by ukiyo-dev** [reply]
Upscale_VL > Bicubic > Upscale_VL > FSR_RCAS 100% 效果如下：
https://imgsli.com/MzU5MjA2

**Reply by chlbc** [reply]
關於 Bicubic 那邊的參數，在 4K 螢幕下似乎可以改選「適應」，並將寬、高皆設成`0.5`，效果應該會一樣，而且不用根據不同遊戲的原生解析度不同而修改配置。Bicubic 部分配置如下：
```json
                {
                    "name": "Bicubic",
                    "scalingType": 1,
                    "scale": {
                        "x": 0.5,
                        "y": 0.5
                    },
                    "parameters": {
                        "paramB": 0.0,
                        "paramC": 0.75
                    }
                },
```

**Comment by sl-shen** [comment]
2k有什么推荐设置吗，还是就跑一次upscale就好

**Reply by XiaofengdiZhu** [reply]
对，标准2k刚好是720p的两倍

**Reply by BangrongMo** [reply]
> 2k有什么推荐设置吗，还是就跑一次upscale就好

建议restore一下，我也是2k屏，感觉直接一次upscale有点糊
